src/statemachine.rs

```diff
@@ -552,12 +568,37 @@ impl StateMachine {
                         }
                     };
 
-                    match makecred.do_pre_flight(&mut dev) {
-                        Ok(()) => { /* Continue with the request */ }
-                        Err(e) => {
-                            warn!("error happened: {e}");
-                            callback.call(Err(e));
-                            break;
+                    // Do "pre-flight": Filter the exclude-list
+                    if dev.get_authenticator_info().is_some() {
+                        makecred.exclude_list = unwrap_result!(
+                            do_credential_list_filtering_ctap2(
+                                &mut dev,
+                                &makecred.exclude_list,
+                                &makecred.rp,
+                                pin_uv_auth_result.get_pin_uv_auth_token(),
+                            ),
+                            callback
+                        );
+                    } else {
+                        let key_handle = do_credential_list_filtering_ctap1(
+                            &mut dev,
+                            &makecred.exclude_list,
+                            &makecred.rp,
+                            &makecred.client_data_hash,
+                        );
+                        // That handle was already registered with the token
+                        if key_handle.is_some() {
+                            // Now we need to send a dummy registration request, to make the token blink
+                            // Spec says "dummy appid and invalid challenge". We use the same, as we do for
+                            // making the token blink upon device selection.
+                            let msg = unwrap_result!(dummy_make_credentials_cmd(), callback);
```

Comment:
`dummy_make_credentials_cmd` can't actually fail here. We should just hard-code a bogus client data hash so that we don't need to handle a failure in the crypto backend.